soccer/MainWindow.cpp

```diff
@@ -718,14 +703,54 @@ void MainWindow::updateStatus() {
     RJ::Time curTime = RJ::now();
 
     // Determine if we are receiving packets from an external referee
-    bool haveExternalReferee =
-        (curTime - ps.lastRefereeTime) < RJ::Seconds(0.5);
+    bool haveExternalReferee = (curTime - ps.lastRefereeTime) < RJ::Seconds(1);
+
+    std::vector<int> validIds = _processor->state()->ourValidIds();
 
-    /*if (_autoExternalReferee && haveExternalReferee &&
-    !_ui.externalReferee->isChecked())
-    {
-        _ui.externalReferee->setChecked(true);
-    }*/
+    for (int i = 1; i < 17; i++) {
+        // This will need to change if the max number of robots changes
+        if (std::find(validIds.begin(), validIds.end(), i - 1) !=
+            validIds.end()) {
+            // The list starts with None so i is 1 higher than the shell id
+            _ui.goalieID->setItemData(i, true, Qt::UserRole);
+        } else {
+            _ui.goalieID->setItemData(i, false, Qt::UserRole - 1);
+        }
+    }
+
+    if (haveExternalReferee && _autoExternalReferee) {
+        // External Ref is connected and should be used
+        _ui.fastHalt->setEnabled(false);
+        _ui.fastStop->setEnabled(false);
+        _ui.fastReady->setEnabled(false);
+        _ui.fastForceStart->setEnabled(false);
+        _ui.fastKickoffBlue->setEnabled(false);
+        _ui.fastKickoffYellow->setEnabled(false);
+    } else {
+        _ui.fastHalt->setEnabled(true);
+        _ui.fastStop->setEnabled(true);
+        _ui.fastReady->setEnabled(true);
+        _ui.fastForceStart->setEnabled(true);
+        _ui.fastKickoffBlue->setEnabled(true);
+        _ui.fastKickoffYellow->setEnabled(true);
+    }
+
+    if (std::find(validIds.begin(), validIds.end(),
+                  _processor->state()->gameState.getGoalieId()) !=
+            validIds.end() &&
```

Comment:
I forgot to mention one other thing:

Right now, if the refbox sends us a invalid ID (not on the field), we use the default (0 in our case). I think that we should set it anyway, because if we set 0 when we aren't supposed to, thats a rule violation. 

A case where this can happen is when we pull the goalie off the field (I'm not sure if this code would reset the ID to 0, which I don't think would happen). But in any case, if we pull the goalie off the field, soccer crashes, and we have to restart, we end up with a goailie ID of 0. 

I think this if statement should be relaxed to be only `(haveExternalRefereee)`. Thoughts?